File: data_structures/stack/ruby/stack.rb
class Stack
    def initialize
        @store = Array.new
    end

    def push(data)
        @store.push(data)
    end

    def pop()
        raise Exception if size == 0
        @store.pop
    end

    def top()
        raise Exception if size == 0
        @store.last
    end

    def size()
        @store.size
    end
end

# Usage
S = Stack.new
S.push(1)
S.push(2)
S.push(3)

puts S.top()
puts S.size()

puts S.pop()
puts S.pop()

File: data_structures/stack/js/stack.js
class Stack {
  constructor () {
    this.data = []
    this.max = -1
  }
  pop () {
    if (this.max === -1)
      throw new Exception("Stack is empty");

    this.max--
  }

  push (data) {
    this.max++
    this.data.push(data)
  }

  top () {
    return this.data[this.max]
  }
}

// usage
const stack = new Stack()
stack.push("a")
stack.push("b")

console.log(stack.top())

stack.pop()

console.log(stack.top())

